src/js/components/pages/scene-editor-gui.jsx

```diff
@@ -0,0 +1,793 @@
+
+var saveTimeout = null;
+
+var SceneEditorGUI = React.createClass({
+
+    getInitialState: function () {
+        return _.extend(this.getStateFromStores(), {
+            focusedMediaObject: null,
+            scene: null,
+            placement: null,
+            mediaObject: null,
+            mediaTypeSupported: null,
+            shouldSave: false,
+            height: 0,
+            width: 0,
+        });
+    },
+
+    getStateFromStores: function () {
+        return {
+            scene: SceneStore.getScene(this.props._id),
+        };
+    },
+
+    componentDidMount: function () {
+        console.log("SceneEditorGUI: Mounted")
+
+        //subscribe to changes
+        SceneStore.addChangeListener(this._onChange);
+        GridStore.addChangeListener(this._onFocusChange)
+
+        //this is the first point where rendered size of area can be captured
+        this.setState({width: this.SceneLayoutArea.offsetWidth, height: this.SceneLayoutArea.offsetHeight})
+    },
+
+    componentWillReceiveProps: function (nextProps) {
+        //same as focus change, same handling procedure, required because selected scene change dosn't appear to reliably fire change event?? 
+        //this._onFocusChange();
+    },
+
+    componentWillUnmount: function () {
+        if (saveTimeout) {
+            //this.saveToScene()
+            clearTimeout(saveTimeout);
+        }    
+        SceneStore.removeChangeListener(this._onChange);
+        GridStore.removeChangeListener(this._onFocusChange)
+    },
+
+    _onChange: function () {
+        //scene changed, clear all state
+        log("SceneEditorGUI: Scene change", this.getStateFromStores())
+        this.setState(this.getStateFromStores());
+    },
+
+    _onFocusChange: function () {
+
+        focusedMediaObject = GridStore.getFocusedMediaObject();
+        scene = SceneStore.getScene(this.props._id);
+
+        var mediaObject = null;
+        var placement = null;
+        var mediaTypeSupported = null;
+
+        //have to use try catch because focusedMediaObject can be out of bounds
+        try {
+            mediaObject = scene.scene[focusedMediaObject]
+        } catch (error) {
+            console.log("SceneEditorGUI: failed to read media object", error);
+        }
+
+        //handle different media types
+        if (mediaObject != null) {
+            switch (mediaObject.type) {
+                case "image":
+                    //load image using loader as may take some time.
+                    mediaTypeSupported = true
+                    placement = this.getPlacementFromMfJSON(mediaObject);
+                    break;
+                case "video":
+                    mediaTypeSupported = true
+                    placement = this.getPlacementFromMfJSON(mediaObject);
+                    break;
+                default:
+                    //unsupported media type
+                    mediaTypeSupported = false
+                    mediaRepresentation = null
```

Comment:
Unused variable mediaRepresentation -> this is the only place in the code that this is used, unsure on its purpose